### PR TITLE
New GUI based on BS4

### PR DESCRIPTION
**Pull request by dorpvom**

### CODE REVIEW

**Comment by jstucke** [comment]
the tabled results of the plugins "cpu architecture" and "crypto hints" seem to have a different font

**Comment by jstucke** [comment]
firmware tags (those assigned during upload) are just plain text

**Comment by jstucke** [comment]
the comments table on the home page is narrower when only containing short comments

**Comment by jstucke** [comment]
if tables with hover contain another table-hover, the hover animation is kinda broken:
- qemu exec (only individual file results)
- file system metadata

**Comment by jstucke** [comment]
in the "Run additional analysis” window, there seems to be a space/etc. missing between the checkboxes and the labels

**Comment by jstucke** [comment]
navbar can still be broken for certain widths if you're an admin and auth is enabled

**Comment by jstucke** [comment]
when auth is enabled, some pages still lead to redirect loops if you lack rights (but I'm pretty sure this isn't new in this pull request)

File: src/web_interface/templates/show_analysis.html [hunk 1]
```diff
@@ -405,123 +504,49 @@ <h4 class="modal-title">Add analysis to file</h4>
...
+            <div class="collapse" id="comments" style="margin:0;">
+                <table class="table table-bordered" valign="center" align="center" border="0" id="comments-table" style="margin-bottom:0;">
+                    <tbody>
+                        {% for comment in firmware.comments|sort_comments %}
+                        <tr>
+                            <td width="20%" valign="bottom">
+                                <div>{{ comment.author }}</div>
+                                <div>{{ comment.time|int|nice_unix_time }}</div>
+                            </td>
+                            <td width="80%" style="margin:0; padding:0;">
```
Comment (jstucke) [inline src/web_interface/templates/show_analysis.html:516]: no `class="p-0 m-0"` in this block?

**Comment by jstucke** [comment]
the fix for the large file tree text seems to affect the header as well

File: src/web_interface/templates/show_analysis.html [hunk 2]
```diff
@@ -342,61 +439,63 @@ <h4 class="modal-title">Add analysis to file</h4>
...
-                <thead>
-                    <tr class="success">
-                        <th class="result">
+                <thead class="thead-light">
+                    <tr>
+                        <th>
                             Comments
                             {# Add Comment Button #}
                             <form action="/comment/{{ firmware.uid }}" style="float: right; margin-right: 5px;">
                                 <button class="btn btn-primary" type="submit">
```
Comment (jstucke) [inline src/web_interface/templates/show_analysis.html:490]: maybe `btn-sm` would make sense here to keep the table header a bit smaller

File: src/web_interface/templates/system_health.html [hunk 3]
```diff
@@ -4,135 +4,126 @@
...
-                                    <td class="active" style="text-align: left; padding:5px">
+                                    <td style="text-align: left; padding:5px">
                                         {{ plugin_name }}
-                                        {%- if plugin_name in analysis_plugin_info %} {{ analysis_plugin_info[plugin_name][3] }}{% endif -%}
+                                        {% if plugin_name in analysis_plugin_info %} {{ analysis_plugin_info[plugin_name][3] }}{% endif %}
                                     </td>
-                                    <td class="{% if plugin_data.queue > 150 %}alert alert-warning{%else%}active{% endif %}" style="text-align: right; padding:5px">
-                                        {% set span_class = " spin" if plugin_data['active'] else "" %}
+                                    <td class="{% if plugin_data.queue > 150 %}table-warning{%else%}{% endif %}" style="text-align: right; padding:5px">
+                                        {% set span_class = " fa-spin" if plugin_data['active'] else "" %}
```
Comment (jstucke) [inline src/web_interface/templates/system_health.html:105]: 👌

**Comment by jstucke** [comment]
manage_users -> roles still looks broken: 
Auswahl_001

**Comment by dorpvom** [comment]
> the tabled results of the plugins "cpu architecture" and "crypto hints" seem to have a different font

I can't verify this. Screenshot or it didn't happen. And make sure to `Strg+0`.

**Comment by dorpvom** [comment]
> the comments table on the home page is narrower when only containing short comments

I don't know how to fix this. I tried a bunch of stuff. Would you like to have a look?

File: src/web_interface/filter.py [hunk 4]
```diff
@@ -363,3 +361,9 @@ def get_unique_keys_from_list_of_dicts(list_of_dicts: List[dict]):
 
 def is_not_manditory_analysis_entry(item):
     return item not in ['analysis_date', 'plugin_version', 'skipped', 'summary', 'system_version', 'tags']
+
+
+def fix_uid_for_collapse_id(uid):
```
[the remark this answers is not in the document]
Comment (dorpvom) [reply src/web_interface/filter.py:371]: or `u_` or `a`. But do we loose a lot by being expressive?

**Comment by jstucke** [comment]
> > the tabled results of the plugins "cpu architecture" and "crypto hints" seem to have a different font
> 
> I can't verify this. Screenshot or it didn't happen. And make sure to `Strg+0`.

For all (upper) result tables the font size is 15px. For the plugins "crypto hints" and "software components" there are additional (lower) tables with font size 1rem

**Comment by jstucke** [comment]
> > the comments table on the home page is narrower when only containing short comments
> 
> I don't know how to fix this. I tried a bunch of stuff. Would you like to have a look?

seems to be because of the class `table-responsive`. If you remove it, it seems to work as expected

File: src/web_interface/templates/find_missing_analyses.html [hunk 5]
```diff
@@ -5,14 +5,13 @@
 
 {% block body %}
```
[the remark this answers is not in the document]
Comment (dorpvom) [reply src/web_interface/templates/find_missing_analyses.html:13]: Done

File: src/web_interface/templates/system_health.html (hunk 3 above)
Comment (dorpvom) [reply src/web_interface/templates/system_health.html:105]: :wink:

**Comment by dorpvom** [comment]
> > > the comments table on the home page is narrower when only containing short comments
> > 
> > 
> > I don't know how to fix this. I tried a bunch of stuff. Would you like to have a look?
> 
> seems to be because of the class `table-responsive`. If you remove it, it seems to work as expected

Jepp

**Comment by dorpvom** [comment]
Re-Review is done. Comments regarding navbar width and auth redirect are tabled to go through another Issue / PR.
Everything else should be done

**Comment by weidenba** [comment]
on some screen resolutions we have overlapping buttons on show analysis view
fact_bs_bug_overlapping_buttons

**Comment by weidenba** [comment]
> 
> 
> on some screen resolutions we have overlapping buttons on show analysis view
> fact_bs_bug_overlapping_buttons

I could write a filter function that  shortens to long badges DOS-Style. It would look like this:
`/_0e0598b137fdf771062d9334802b8caed5c67b7e576e612b2821a533becfbbc5_21162492.extracted/2DF800.squashfs` -> `/_0e0598b137fdf771062d9334802b8caed5c6~/2DF800.squashfs`

**Pull request merged**